output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/io/ValidationReport.java

```diff
@@ -0,0 +1,154 @@
+   *     json string passed as parameter.
+   */
+  public static ValidationReport fromJsonString(String jsonString) {
+    return GSON.fromJson(jsonString, ValidationReport.class);
+  }
+
+  /**
+   * Returns the list of {@code NoticeSummary} of this {@code ValidationReport}.
+   *
+   * @return the list of {@code NoticeSummary} of this {@code ValidationReport}.
+   */
+  public List<NoticeSummary> getNotices() {
+    return notices;
+  }
+
+  /**
+   * Returns the immutable and ordered set of error codes contained in this {@code ValidationReport}
+   *
+   * @return the immutable and ordered set of error codes contained in this {@code ValidationReport}
+   */
+  public ImmutableSet<String> getErrorCodes() {
+    return errorCodes;
+  }
+
+  /**
+   * Compares two validation reports: returns true if they contain the same set of error codes.
+   *
+   * @param otherValidationReport the other {@code ValidationReport}.
+   * @return true if the two {@code ValidationReport} contain the same set of error codes, false
+   *     otherwise.
+   */
+  public boolean hasSameErrorCodes(ValidationReport otherValidationReport) {
+    return getErrorCodes().equals(otherValidationReport.getErrorCodes());
+  }
+
+  /**
+   * Returns the number of new error codes introduced by the other {@code ValidationReport} passed
+   * as parameter, e.g. if this {@code ValidationReport} has the following error codes:
+   *
+   * <ul>
+   *   <li>invalid_phone_number;
+   *   <li>number_out_of_range;
+   * </ul>
+   *
+   * and the other {@code ValidationReport} has the following error codes:
+   *
+   * <ul>
+   *   <li>invalid_phone_number;
+   *   <li>number_out_of_range;
+   *   <li>invalid_email_address;
+   *   <li>invalid_url;
+   * </ul>
+   *
+   * <p>then this methods returns 2 as it contains two new errors codes (invalid_email_address,
+   * invalid_url) not present in this {@code ValidationReport}
+   *
+   * @param other the other {@code ValidationReport}
+   * @return the number of new error codes introduced by the other {@code ValidationReport} passed
+   *     as parameter.
+   */
+  public int getNewErrorCount(ValidationReport other) {
+    return Sets.difference(other.getErrorCodes(), getErrorCodes()).size();
+  }
+
+  /**
+   * Determines if two validation reports are equal regardless of the order of the fields in the
+   * list of {@code NoticeSummary}.
+   *
+   * @param other the other {@code ValidationReport}.
+   * @return true if both validation reports are equal regardless of the order of the fields in the
+   *     list of {@code NoticeSummary}.
+   */
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (other instanceof ValidationReport) {
+       return getNotices().size() == ((ValidationReport) other).getNotices().size() &&
+      getNotices().containsAll(((ValidationReport) other).getNotices());
```

Comment:
This indentation looks unusual, could you apply automatic formatting for the whole PR?

The containsAll() method here will be O(n*2) since it works on Lists. Do you mind to use hash sets instead and reduce the complexity to O(n)? You will need to provide a hash code function for NoticeSummary class.